src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -90,10 +90,12 @@ public T remove(int index) {
 
     @Override
     public boolean remove(T t) {
-        for (int i = 0; i < size; i++) {
-            if (t == getNode(i).value || (t != null && getNode(i).value.equals(t))) {
-                remove(i);
-                return true;
+        if (!isEmpty()) {
+            for (int i = 0; i < size; i++) {
+                if (t == getNode(i).value || (t != null && getNode(i).value.equals(t))) {
+                    remove(i);
+                    return true;
```

Comment:
```suggestion
            for (int i = 0; i < size; i++) {
                if (t == getNode(i).value || (getNode(i).value != null && getNode(i).value.equals(t))) {
                    remove(i);
                    return true;
```
is empty check doesn't fix problem, if condition was wrong